wrapspawner/wrapspawner.py

```diff
@@ -212,6 +213,14 @@ class ProfilesSpawner(WrapSpawner):
             the first item starts selected."""
         )
 
+    def __init__(self, *args, **kwargs):
+        super(ProfilesSpawner, self).__init__(*args, **kwargs)
+
+        keys = [p[1] for p in self.profiles]
```

Comment:
The typical place for code like this is a traits `@validate` method, rather than `__init__`:

```python
from traitlets import validate

@validate("profiles")
def _validate_profiles(self, proposal):
    profiles = proposal.value
    # any checks, warnings, errors or coercions go here

    # return the value
    return profiles
```

Reply:
Thanks for the feedback. Would you rather prefer to raise a `TraitError` or simply log a warning message ?

Reply:
I think an error is probably appropriate, but I'll leave it up to you.

Reply:
Hum, ok thanks. I also believe that an error is appropriate as otherwise, we could have a non-working configuration which would fail without any error.